### PR TITLE
python3Packages.rio-tiler: init at 6.6.1

### PR DESCRIPTION
**Pull request by sikmir**
## Description of changes
[**rio-tiler**](https://github.com/cogeotiff/rio-tiler) - user friendly Rasterio plugin to read raster datasets.

Added missed dependencies: morecantile, rioxarray, color-operations, pystac.

Closes #287477 (@frafra)

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 312708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.pystac</li>
    <li>python311Packages.pystac.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.pystac</li>
    <li>python312Packages.pystac.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 312708` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.color-operations</li>
    <li>python311Packages.color-operations.dist</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.pystac</li>
    <li>python311Packages.pystac.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python312Packages.color-operations</li>
    <li>python312Packages.color-operations.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.pystac</li>
    <li>python312Packages.pystac.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/morecantile/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ lib
```
Comment (imincik) [inline pkgs/development/python-modules/morecantile/default.nix:1]: I suggest to alphabetically reorder function parameters and visually split them to functions and packages as we have in [other packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/rasterio/default.nix).

File: pkgs/development/python-modules/pystac/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{ lib
```
Comment (imincik) [inline pkgs/development/python-modules/pystac/default.nix:1]: I suggest to alphabetically reorder function parameters and visually split them to functions and packages as we have in [other packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/rasterio/default.nix).

File: pkgs/development/python-modules/morecantile/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    rev = version;
+    hash = "sha256-F7xYQrOngoRsZjmS6ZHRGN0/GD53AYcMQzyY1LZ1O7I=";
+  };
+
+  nativeBuildInputs = [ flit ];
+
+  propagatedBuildInputs = [ attrs click pydantic pyproj ];
+
+  nativeCheckInputs = [ mercantile pytestCheckHook rasterio ];
+
```
Comment (imincik) [inline pkgs/development/python-modules/morecantile/default.nix:34]: Is it possible to add `pythonImportsCheck` here ?

File: pkgs/development/python-modules/pystac/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+buildPythonPackage rec {
+  pname = "pystac";
+  version = "1.10.1";
+  pyproject = true;
+  disabled = pythonOlder "3.9";
+
+  src = fetchFromGitHub {
+    owner = "stac-utils";
+    repo = "pystac";
+    rev = "v${version}";
```
Comment (imincik) [inline pkgs/development/python-modules/pystac/default.nix:26]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/color-operations/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    repo = "color-operations";
+    rev = version;
+    hash = "sha256-KsrgilcNK2ufPKrhtGdf8mdlFzhsHB2jHN+WDlZqabc=";
+  };
+
+  nativeBuildInputs = [ cython setuptools ];
+
+  propagatedBuildInputs = [ oldest-supported-numpy ];
+
+  doCheck = false;
```
Comment (imincik) [inline pkgs/development/python-modules/color-operations/default.nix:30]: Why disabling `doCheck` is needed here ?

File: pkgs/development/python-modules/color-operations/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{ lib
```
Comment (imincik) [inline pkgs/development/python-modules/color-operations/default.nix:1]: I suggest to alphabetically reorder function parameters and visually split them to functions and packages as we have in [other packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/rasterio/default.nix).

File: pkgs/development/python-modules/rioxarray/default.nix [hunk 7]
```diff
@@ -0,0 +1,56 @@
+{ lib
```
Comment (imincik) [inline pkgs/development/python-modules/rioxarray/default.nix:1]: I suggest to alphabetically reorder function parameters and visually split them to functions and packages as we have in [other packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/rasterio/default.nix).

File: pkgs/development/python-modules/rioxarray/default.nix [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    owner = "corteva";
+    repo = "rioxarray";
+    rev = version;
+    hash = "sha256-bumFZQktgUqo2lyoLtDXkh6Vv5oS/wobqYpvNYy7La0=";
+  };
+
+  build-system = [ setuptools ];
+
+  propagatedBuildInputs = [
+    packaging
```
Comment (imincik) [inline pkgs/development/python-modules/rioxarray/default.nix:33]: Please reorder this list alphabetically. Thanks.

File: pkgs/development/python-modules/rio-tiler/default.nix [hunk 9]
```diff
@@ -0,0 +1,64 @@
+{ lib
```
Comment (imincik) [inline pkgs/development/python-modules/rio-tiler/default.nix:1]: I suggest to alphabetically reorder function parameters and visually split them to functions and packages as we have in [other packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/rasterio/default.nix).

File: pkgs/development/python-modules/rio-tiler/default.nix [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    owner = "cogeotiff";
+    repo = "rio-tiler";
+    rev = version;
+    hash = "sha256-MR6kyoGM3uXt6JiIEfGcsmTmxqlLxUF9Wn+CFuK5LtQ=";
+  };
+
+  build-system = [ hatchling ];
+
+  propagatedBuildInputs = [
+    attrs
```
Comment (imincik) [inline pkgs/development/python-modules/rio-tiler/default.nix:38]: Please, reorder this list alphabetically.

File: pkgs/development/python-modules/rio-tiler/default.nix [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    numpy
+    morecantile
+    pydantic
+    pystac
+    rasterio
+    color-operations
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
```
Comment (imincik) [inline pkgs/development/python-modules/rio-tiler/default.nix:51]: Please, reorder this list alphabetically.

File: pkgs/development/python-modules/color-operations/default.nix [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+, pythonOlder
+, pytestCheckHook
+, colormath
+, cython
+, oldest-supported-numpy
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "color-operations";
```
Comment (imincik) [inline pkgs/development/python-modules/color-operations/default.nix:14]: What about changing the package name to `rio-color-operations` ?

File: pkgs/top-level/python-packages.nix [hunk 13]
```diff
@@ -2360,6 +2360,8 @@ self: super: with self; {
 
   collidoscope = callPackage ../development/python-modules/collidoscope { };
 
+  color-operations = callPackage ../development/python-modules/color-operations { };
```
Comment (imincik) [inline pkgs/top-level/python-packages.nix:2363]: What about changing the package name to `rio-color-operations` ?

**Review by imincik — CHANGES_REQUESTED** [review]
Great work @sikmir , thank you very much. I left few minor comments.

File: pkgs/development/python-modules/pystac/default.nix (hunk 4 above)
Comment (sikmir) [reply pkgs/development/python-modules/pystac/default.nix:26]: Why? prefix required here https://github.com/stac-utils/pystac/tags

File: pkgs/top-level/python-packages.nix (hunk 13 above)
Comment (sikmir) [reply pkgs/top-level/python-packages.nix:2361]: > Lightweight version of [rio-color](https://github.com/mapbox/rio-color) but removing rasterio dependency

`color-operations` is not a rasterio plugin.

File: pkgs/development/python-modules/color-operations/default.nix (hunk 5 above)
Comment (sikmir) [reply pkgs/development/python-modules/color-operations/default.nix:34]: Tests failed on self import:
```sh
E   ModuleNotFoundError: No module named 'color_operations.colorspace'
```
I'm not sure how to fix it right now.

File: pkgs/development/python-modules/pystac/default.nix (hunk 4 above)
Comment (imincik) [reply pkgs/development/python-modules/pystac/default.nix:27]: Sorry, I somehow overlooked it. Your are right.

File: pkgs/top-level/python-packages.nix (hunk 13 above)
Comment (imincik) [reply pkgs/top-level/python-packages.nix:2361]: OK. I agree.

File: pkgs/development/python-modules/color-operations/default.nix (hunk 5 above)
Comment (imincik) [reply pkgs/development/python-modules/color-operations/default.nix:34]: Maybe add a comment. I think this is my last suggestion to this PR. We can merge after this.

Comment (sikmir) [reply pkgs/development/python-modules/color-operations/default.nix]: I've found solution how to fix tests:
```nix
  preCheck = ''
    python setup.py build_ext --inplace
  '';
```

**Review by imincik — APPROVED** [review]
Looks good to me now. Great work, thank you very much !

**Pull request merged**